### ISSUE TITLE
Add customisation options such as indentation from 2 to 4, etc in IntelliJ Plugin settings

### ISSUE THREAD

**Issue by kuondream** [google/google-java-format#335]
I tried to search in the issue tracker and also online on how to customise google-java-format intelliJ plugin but I was unable to find an option or alternative to fix it. I even tried download the .xml file in the README.md from this repository and edit the indentation options for the various files changing from 2 to 4 and 4 to 8.

I understand that the main purpose of the plugin is to support google coding style format. I observe that there is also a drop down option in the plugin to choose the styles currently there are only two styles with no option to add new styles. I find this plugin very useful but I was hoping you could consider adding extra customisation options as for example in some cases, some coding styles follow exactly the same as google style format except for indentations but despite trying to change the indentations settings manually in the code style section under settings in intelliJ, google-java-format seem to overwrite it to 2 indentations.

I believe it would be extremely useful to add some small customisations such as this for situations where some coding styles are almost identical to google style format except the indentations settings.

**Comment by tbroyer** [comment]
From the README:
> _**Note:** There is no configurability as to the formatter's algorithm for formatting. This is a deliberate design decision to unify our code formatting on a single format._

So I'd guess the answer would be "no".

You could possibly create a fork for your own style guide though.

**Comment by kuondream** [comment]
@tbroyer Yeah I read that part I did look through the source code. I might try coding a forked version myself in the future as you suggested. I was just hoping maybe there could be at least small changes allowed. I understand the part about unifying everything to a single format but even so I was hoping that it may be allowed an option whether a not to fully override or being given priority over some settings that is more susceptible to slight changes such as indentations.

I still appreciate the reply though. Thank you.
